utils/wasm-builder/Cargo.toml

```diff
@@ -13,13 +13,28 @@ homepage = "https://substrate.io"
 targets = ["x86_64-unknown-linux-gnu"]
 
 [dependencies]
-ansi_term = "0.12.1"
-build-helper = "0.1.1"
-cargo_metadata = "0.14.2"
-strum = { version = "0.24.1", features = ["derive"] }
-tempfile = "3.1.0"
-toml = "0.5.4"
-walkdir = "2.3.2"
-sp-maybe-compressed-blob = { version = "4.1.0-dev", path = "../../primitives/maybe-compressed-blob" }
-filetime = "0.2.16"
-wasm-opt = "0.110"
\ No newline at end of file
+ansi_term = { version = "0.12.1", optional = true }
+build-helper = { version = "0.1.1", optional = true }
+cargo_metadata = { version = "0.14.2", optional = true }
+strum = { version = "0.24.1", features = ["derive"], optional = true }
+tempfile = { version = "3.1.0", optional = true }
+toml = { version = "0.5.4", optional = true }
+walkdir = { version = "2.3.2", optional = true }
+sp-maybe-compressed-blob = { version = "4.1.0-dev", path = "../../primitives/maybe-compressed-blob", optional = true }
+filetime = { version = "0.2.16", optional = true }
+wasm-opt = { version = "0.110", optional = true }
+
+[features]
+default = ["std"]
```

Comment:
Good idea, but way too complicated. Just adding target dependencies with `not(target_arch = "wasm32")` should do it.
Then we also don't need to enable or disable any features. 

https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html#platform-specific-dependencies

Reply:
I tried that initially, the problem is that `substrate-wasm-builder` itself is never compiled for `wasm32` architecture, it is always compiled to native machine code. The cleanest way to work around that that I found was to add an `std` feature.

I was hoping there is a way to distinguish host from target, but apparently not in this context.

Reply:
Good point, but you should be able to import wasm-builder conditionally in your downstream project based on the target_arch? Achieving the same without requiring any upstream changes?

Reply:
Almost, `provide_dummy_wasm_binary_if_not_exist` is still necessary for WASM, so I thought having it upstream in a backwards compatible way would be more convenient than reimplementing that function in every runtime.

Reply:
Why is that required? The wasm binary is only imported on std builds?

Reply:
Indeed. I recall there was something when I was testing, will check again later.